src/components/EventForm/EventForm.tsx

```diff
@@ -0,0 +1,186 @@
+import React, { useState, useCallback } from 'react';
+import { RouteComponentProps } from '@reach/router';
+import { formatISO, format } from 'date-fns';
+
+import Alert from 'src/components/Alert';
+import Button from 'src/components/inputs/Button';
+import DateField from 'src/components/fields/DateField';
+import SelectField from 'src/components/fields/SelectField';
+import TextAreaField from 'src/components/fields/TextAreaField';
+import TextField from 'src/components/fields/TextField';
+import TimeField from 'src/components/fields/TimeField';
+import { EventFormWrapper, Form } from './styled';
+import { EVENT_OPTIONS } from './constants';
+
+export type EventDataInput = {
+  id?: string;
+  name?: string;
+  description?: string;
+  startTime?: Date;
+  endTime?: Date;
+  imageUrl?: string;
+  imageAlt?: string;
+  location?: string;
+  eventType?: string;
+};
+
+export type EventDataOutput = {
+  id?: string;
+  name: string;
+  description: string;
+  startTime: Date;
+  endTime: Date;
+  imageUrl?: string;
+  imageAlt?: string;
+  location?: string;
+  eventType: string;
+};
+
+type EventFormProps = {
+  headerTitle: string;
+  onSubmit: (data: EventDataOutput) => Promise<void>;
+} & EventDataInput;
+
+function EventForm({ onSubmit, ...rest }: EventFormProps) {
+  const [name, setName] = useState(rest.name || '');
+  const [imageUrl, setImageUrl] = useState(rest.imageUrl || '');
+  const [imageAlt, setImageAlt] = useState(rest.imageAlt || '');
+  const [date, setDate] = useState(
+    format(
+      new Date((rest.startTime || new Date(Date.now())).toDateString()) ||
+        Date.now(),
+      'yyyy-MM-dd'
+    )
+  );
+  const [startTime, setStartTime] = useState(
+    format(rest.startTime || Date.now(), 'HH:mm')
+  );
+  const [endTime, setEndTime] = useState(
+    format(rest.endTime || Date.now(), 'HH:mm')
+  );
+  const [description, setDescription] = useState(rest.description || '');
+  const [eventType, setEventType] = useState(rest.eventType || '');
+  const [location, setLocation] = useState(rest.location || '');
+
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [hasSubmitted, setHasSubmitted] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+
+  const isFormValid = name && description && eventType;
+  const isSubmitDisabled = !isFormValid || isSubmitting;
+
+  const handleSubmit = useCallback(async () => {
+    setIsSubmitting(true);
+    setError(null);
+
+    try {
+      await onSubmit({
+        id: rest.id,
+        name,
+        description,
+        startTime: new Date(date + ' ' + startTime),
+        endTime: new Date(date + ' ' + endTime),
```

Comment:
See my other comment about native Date methods vs. functions from `date-fns`.

I see that we're also doing this in the CreateEventPage. If we change this part then we should also apply the same change to CreateEventPage.